services/ui_backend_service/api/search.py

```diff
@@ -0,0 +1,268 @@
+import asyncio
+from typing import Tuple
+from services.data.db_utils import translate_run_key
+from services.data.tagging_utils import apply_run_tags_to_db_response
+from services.utils import handle_exceptions, logging
+from services.ui_backend_service.data.cache.utils import (
+    search_result_event_msg, error_event_msg
+)
+from .utils import query_param_enabled
+from urllib.parse import unquote_plus
+
+from aiohttp import web
+import json
+
+SCOPE_ARTIFACT = 'ARTIFACT'
+SCOPE_FOREACH_VARIABLE = 'FOREACH_VARIABLE'
+
+
+class SearchApi(object):
+    def __init__(self, app, db, cache=None):
+        self.db = db
+        app.router.add_route(
+            "GET", "/search/flows/{flow_id}/runs/{run_number}", self.get_run_tasks
```

Comment:
yeah, for more context, we had to change the api because we internally websocket whitelist has a prefix matching rule, so we can only enable `ws:.../search` but not `ws:.../{flow_info}/search`